File: java_project/e_librarian_api/src/main/java/com/elibrarian/demo/services/UsersService.java
package com.elibrarian.demo.services;

import java.util.List;
import java.util.Optional;

import com.elibrarian.demo.model.UsersEntity;

public interface UsersService {
    UsersEntity addUser(UsersEntity user);

    List<UsersEntity> viewUsers();

    Optional<UsersEntity> viewUserById(Integer uId);

    Optional<UsersEntity> viewUserByUsername(String username);

    UsersEntity updateUser(UsersEntity user);

    Integer removeUser(Integer uId);
}

File: elibrarian/src/main/java/com/elibrarian/service/UsersService.kt
package com.librarian.elibrarian.service

import com.librarian.elibrarian.model.Users
import java.util.*

interface UsersService {
    fun create(users: Users): Users
    fun getAll(): List<Users>
    fun getById(uId: Int): Optional<Users>
    fun getByUsername(username: String): Optional<Users>
    fun update(users: Users): Users
    fun remove(uId: Int): Int
}